packages/amplify-category-api/commands/api/add-graphql-datasource.js

```diff
@@ -106,15 +107,16 @@ module.exports = {
         const apiDirPath = `${projectBackendDirPath}/${category}/${resourceName}`;
         fs.ensureDirSync(apiDirPath);
         const graphqlSchemaFilePath = `${apiDirPath}/schema.graphql`;
-        fs.ensureFileSync(graphqlSchemaFilePath);
-        const graphqlSchemaRaw = fs.readFileSync(graphqlSchemaFilePath, 'utf8');
-        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
-
         const rdsGraphQLSchemaDoc = graphqlSchemaContext.schemaDoc;
 
+        // to check for both schema.graphql file or schema directory and dump RDS schema
+        const graphqlSchemaRaw = readSchema(apiDirPath, rdsGraphQLSchemaDoc);
```

Comment:
From the code it looks like the schema is read and if its inside a `schema` directory, a new file called `rds.graphql` is written to directory. The functions should have single resposiblilty and should not change their behavior of reading/writing based on folder structure.

Reply:
Fixed